pkg/controllers/user/controller.go

```diff
@@ -288,6 +286,12 @@ func (h *handler) OnSecretChange(key string, secret *v1.Secret) (*v1.Secret, err
 	}
 	token := string(secret.Data["token"])
 
+	contextName := h.cfg.ContextName
+	user, err := getUserByName(userName, h)
+	if err == nil && user.Spec.Context != "" {
+		contextName = user.Spec.Context
+	}
```

Comment:
Idea: if the `context` field on the user is not set, should klum populate it? Kind of like how pods automatically apply fields like dnsPolicy, serviceAccount, etc?

Because if you just update the fallback context name in the klum config, does it also regenerate kubeconfigs with the new fallback context name?

Reply:
I like the idea. Would this be sufficient? [101dc35](https://github.com/jadolg/klum/pull/93/commits/101dc352812ea782f98f29c504859f147de8029b)

Reply:
LGTM good to me!